fulp_modules/main_features/bloodsuckers/code/bloodsucker/bloodsucker_life.dm

```diff
@@ -330,27 +346,28 @@
 			if(owner.current.AmStaked())
 				to_chat(owner.current, "<span class='userdanger'>You are staked! Remove the offending weapon from your heart before sleeping.</span>")
 				return
-			/// Otherwise, check for Sol, or if injured enough to enter Torpor.
-			if(clan.bloodsucker_sunlight.amDay || total_damage >= 10)
-				to_chat(owner.current, "<span class='notice'>You enter the horrible slumber of deathless Torpor. You will heal until you are renewed.</span>")
-				Torpor_Begin()
-	/// If it's not Sol and you have 0 Brute damage, check to End Torpor.
-	if(!clan.bloodsucker_sunlight.amDay && total_brute <= 10)
-		if(HAS_TRAIT(owner.current, TRAIT_NODEATH))
+			/// Otherwise, check if it's Sol, to enter Torpor.
+			if(clan.bloodsucker_sunlight.amDay)
+				Check_Begin_Torpor(TRUE)
+		/// You are in Torpor, and in a Coffin. Check if it's not Daytime & you have less than 10 Brute/Burn combined to end Torpor.
+		else if(!clan.bloodsucker_sunlight.amDay && total_damage <= 10)
 			Check_End_Torpor()
-
-/datum/antagonist/bloodsucker/proc/Torpor_Begin()
-	/// Force them to go to sleep
-	REMOVE_TRAIT(owner.current, TRAIT_SLEEPIMMUNE, BLOODSUCKER_TRAIT)
-	/// Without this, you'll just keep dying while you recover.
-	ADD_TRAIT(owner.current, TRAIT_NODEATH, BLOODSUCKER_TRAIT)
-	ADD_TRAIT(owner.current, TRAIT_FAKEDEATH, BLOODSUCKER_TRAIT)
-	ADD_TRAIT(owner.current, TRAIT_DEATHCOMA, BLOODSUCKER_TRAIT)
-	owner.current.Jitter(0)
-	/// Disable ALL Powers
-	for(var/datum/action/bloodsucker/power in powers)
-		if(power.active && !power.can_use_in_torpor)
-			power.DeactivatePower()
+	/// You're not in a Coffin, but are in Torpor. Check if it's not Daytime, & you have less than 10 Brute (NOT Burn) to end Torpor.
+	else if(!clan.bloodsucker_sunlight.amDay && total_brute <= 10 && HAS_TRAIT(owner.current, TRAIT_NODEATH))
+		Check_End_Torpor()
+
+/datum/antagonist/bloodsucker/proc/Check_Begin_Torpor(SkipChecks = FALSE)
+	/// Are we entering Torpor via Sol/Death? Then entering it isnt optional!
+	if(SkipChecks)
+		Torpor_Begin()
+		return
+	var/mob/living/carbon/user = owner.current
+	var/total_brute = user.getBruteLoss_nonProsthetic()
+	var/total_burn = user.getFireLoss_nonProsthetic()
+	var/total_damage = total_brute + total_burn
+	/// Checks - Not daylight & Has more than 10 Brute/Burn & not already in Torpor
+	if(!clan.bloodsucker_sunlight.amDay && total_damage >= 10 && !HAS_TRAIT(owner.current, TRAIT_NODEATH))
+		Torpor_Begin()
 
 /datum/antagonist/bloodsucker/proc/Check_End_Torpor()
```

Comment:
Check_End_Torpor should probably have SkipChecks like Check_Begin_Torpor since it is often called when the variables that end torpor are already evaluated, like on 357 and 354

Note that Check_End_Torpor() will end torpor whenever it's called if you're not in a coffin, since 377 does not check for brute damage as of the time of review

This will still function correctly since Check_End_Torpor is only called outside of a coffin in HandleTorpor when you have less than 10 brute, but it'd be good to have the checks standardized

Reply:
Check_End_Torpor cancels out if you're not in Torpor already.
if(!HAS_TRAIT(owner.current, TRAIT_NODEATH))
		return